### DISCUSSION TITLE
adCoupledScalarDot problem in ADScalarTimeKernel

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#25655] [Q&A General] [answered]
Hi all,

I would like to develop a CoupledADScalarTimeDerivative which inherits ADScalarTimeKernel. 
But ADScalarTimeKernel doesn't have the adCoupledScalarDot function. I can not define the variable like:
 **_v_dot(adCoupledScalarDot("v"))**

I also find CoupledODETimeDerivative defining **_v_dot(coupledScalarDot("v"))**, but it is not AD type. How to define an adCoupledScalarDot variable in ADScalarTimeKernel?

Thank you!

**Comment by GiudGiud** [comment]
@joshuahansel do you remember why we dont have that one

**Reply by joshuahansel** [reply]
I don't believe there is any reason - we just need to add `adCoupledScalarDot` to `ScalarCoupleable`.

**Reply by SomnusYu** [reply]
So, could you please tell me what should I do?

**Reply by GiudGiud** [reply, marked as answer]
You ll need to edit the ScalarCoupleable.C/.h files in the src/include / interfaces folder(s) to add this new routine

**Reply by SomnusYu** [reply]
Ok, I'll have a try. Thanks!

**Reply by SomnusYu** [reply]
Hi,

I have added the function to ScalarCoupleable:

ScalarCoupleable.h:
 `const ADVariableValue & adCoupledScalarDot(const std::string & var_name, unsigned int comp = 0) const;` 

ScalarCoupleable.C:
```
const ADVariableValue &
ScalarCoupleable::adCoupledScalarDot(const std::string & var_name, unsigned int comp) const
{
  checkVar(var_name);
  validateExecutionerType(var_name, "adCoupledScalarDot");
  return getScalarVar(var_name, comp)->adUDot();
}
```
The compilation is passed. But I'm not sure if it is OK. Is there anything I still need to add? or that is enough?

**Reply by GiudGiud** [reply]
looks good to me

**Reply by SomnusYu** [reply]
Thanks!
